images.go

```diff
@@ -478,9 +484,14 @@ func (r *imageStore) SetMetadata(id, metadata string) error {
 }
 
 func (r *imageStore) removeName(image *Image, name string) {
+	image.addNameToHistory(name)
```

Comment:
`removeName` is, AFAICS, only called on a _name conflict_ (e.g. if `SetNames` adds another name to an image, and that name has been used by a different image); it is not called on _untag_ (`SetNames` removing an existing name).

That’s good enough as far as the original motivation goes:
> I think the major use case is: Displaying the image name to the user if an image has no Names any more due to re-tag, point 4

but not consistent with how `NamesHistory` is documented:
> `NamesHistory is an optional set of Names the image had in the past. `

Reply:
Alright, I updated the documentation as suggested.

Reply:
I was somewhat expecting an implementation change instead of documenting the behavior, but, sure, this is consistent. Up to @nalind.

Reply:
Yeah I can also change the implementation to match the previous documentation, this was not clear for me in the first place.

Reply:
Changed the implementation and added tests on top.